src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -224,9 +224,22 @@ bool canInferencePadsForNNPAConv(ONNXConvOp op) {
   ONNXConvOpAdaptor operandAdaptor = ONNXConvOpAdaptor(op);
   ONNXConvOpShapeHelper shapeHelper(&op);
   assert(succeeded(shapeHelper.computeShape(operandAdaptor)));
-  return (shapeHelper.pads.size() == 4) &&
-         (llvm::all_of(
-             shapeHelper.pads, [](IndexExpr val) { return val.isLiteral(); }));
+  RankedTensorType inputType = op.X().getType().cast<RankedTensorType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+  // dimension of inferenced pads should be 4D
+  if (shapeHelper.pads.size() != 4)
+    return false;
+  // all dimensions of pad should be literal
+  if (llvm::any_of(
```

Comment:
@negiyas Are these constrains (no valid, image size being compile time constant) needed because there is a bug otherwise, or it is a real constraints, namely we cannot use CPU padding in such case?

Naively, I would have assumed that we can always do CPU padding.

This PR is fine, I would like to understand if long term we should have a better fix.